core/src/test/scala/unit/kafka/security/auth/SimpleAclAuthorizerTest.scala

```diff
@@ -271,4 +359,46 @@ class SimpleAclAuthorizerTest extends ZooKeeperTestHarness {
 
     acls
   }
+
+  /**
+    * https://github.com/junit-team/junit/wiki/Multithreaded-code-and-concurrency
+    *
+    * To use this you pass in a Collection of Runnables that are your arrange\act\assert test on the SUT,
+    * they all run at the same time in the assertConcurrent method; the chances of triggering a multithreading code error,
+    * and thereby failing some assertion are greatly increased.
+    */
+  private def assertConcurrent(message: String, runnables: List[Runnable], maxTimeoutSeconds: Int) {
+    val numThreads = runnables.size
+    val exceptions = Collections.synchronizedList(new util.ArrayList[Throwable]())
+    val threadPool = Executors.newFixedThreadPool(numThreads)
+    try {
+      val allExecutorThreadsReady = new CountDownLatch(numThreads)
+      val afterInitBlocker = new CountDownLatch(1)
+      val allDone = new CountDownLatch(numThreads)
+      for (submittedTestRunnable <- runnables) {
+        threadPool.submit(new Runnable() {
+          def run() {
+            allExecutorThreadsReady.countDown()
+            try {
+              afterInitBlocker.await()
+              submittedTestRunnable.run()
+            } catch {
+              case e: Throwable => exceptions.add(e)
+            } finally {
+              allDone.countDown()
+            }
+          }
+        })
+      }
```

Comment:
@ijuma I understand the change may be better. But I have just spent 7 days fixing an issue that blocks the real patch (#1005) I would like to get in. I have a lot more to work on before 0.10 and would like to spend my time there. This code works, it proves the fix works, and its test code...so we can always make it better.